src/sanitize-argument.ts

```diff
@@ -107,6 +122,30 @@ export function sanitize_redis(argv:any)
   }
 }
 
+export async function checkApiAvailabilty(url: string): Promise<boolean>{
+  try {
+    const resp = await axios.get(url, { maxRedirects: 0 });
+    return resp.status === 200 && !resp.headers['mediawiki-api-error'];
+  } catch (err) {
+    return false;
+  }
+}
+
+export async function sanitize_customMainPage(customMainPage: string, restApiPath: string, base: string, mwApiPath: string, noLocalParserFallback: boolean) {
+  const baseUrl = new URL(ensureTrailingChar(base, '/'));
+  const apiPath = mwApiPath ?? 'w/api.php';
+  const apiUrl = new URL(`${apiPath}?`, baseUrl);
+  const veApiUrl = new URL(`${apiUrl.href}action=visualeditor&mobileformat=html&format=json&paction=parse&page=`);
+  const desktopRestApiUrl = new URL(ensureTrailingChar(new URL(restApiPath ?? 'api/rest_v1/page/html', baseUrl.href).toString(), '/'));
+  const baseUrlForMainPage = checkApiAvailabilty(desktopRestApiUrl.href + encodeURIComponent(customMainPage)) ? desktopRestApiUrl.href :
+                             checkApiAvailabilty(veApiUrl.href) ? veApiUrl.href :
+                             undefined;
+  if (!checkApiAvailabilty(baseUrlForMainPage + encodeURIComponent(customMainPage)) && noLocalParserFallback) {
+    throw new Error(`Failed to verify status code 200 on custom main page`)
+  }
+  return customMainPage.replace(/ /g,'_')
+}
```

Comment:
In  #1318, I just ask to test if the `customMainPage` article exists at all. I'm sure we can do a lot simpler. Just checking if `baseUrl + customMainPage` gives HTTP 200 should be sufficient, shouldn't it!?

Reply:
ok